engine/pom.xml

```diff
@@ -218,8 +218,10 @@
                             org.hibernate.search.query;version="${project.version}",
                             org.hibernate.search.query.collector;version="${project.version}",
                             org.hibernate.search.query.dsl;version="${project.version}",
+                            org.hibernate.search.query.dsl.sort;version="${project.version}",
                             org.hibernate.search.query.engine;version="${project.version}",
                             org.hibernate.search.query.engine.spi;version="${project.version}",
+                            org.hibernate.search.query.engine.spi.impl;version="${project.version}",
```

Comment:
We use the "impl" package for anything which is *not* meant to be accessed
by outside code, be it users or integrators of the library. We reserve the
right to change it at will without previous notice. Hence "impl" should
never be exported. We use the "spi" package in order to make specific
behaviours pluggable (mainly for integrators, sometimes also for users).

So usually an SPI is not implemented by ourselves (apart from the default
implementation) but an implementation is provided by users/integrators. In
the case where we provide some base class or utilities helping SPI
implementors to do their job, this one should be part of "spi" as well
(because if the intended usage of these classes is as described, they are
meant for public usage, thus cannot easily be changed as we like and thus
should not be located in "impl").

Of course code in "impl" can refer to code in "spi" or the user API as
needed. Only public code should not expose impl code (by means of usages in
method signatures or inheritance relationships).